### PR TITLE
Add `heading` option to match custom headings

### LINKED ISSUE #12: Add `heading` option to match custom headings
### Subject of the feature

Add a simple `match` option, so one can match any other heading than the english word "license" or "licence".

### Problem

Currently you can't use `remark-license` on multilingual (or just any other language than english) readmes. This would fix it, but probably other use cases exist as well. For example, one could prefer a heading like "License Information" or "Licensing".

### Expected behavior

The regular expression passed to this option should determine the heading under which `remark-license` generates the license info.

### Alternatives

The only alternative solution to this would be to use your own plugin, which is what I considered at first. But I think this can't be a just a "me-issue".

### PR DESCRIPTION
**Pull request by nnmrts**
Closes #12.

Most of this is described in the respective issue. Motivation behind this was that I basically have a multilingual readme setup but `remark-license` only works on english readmes. I'm not sure the name of the option is perfect. I'm fine with `match`, `regexp` and `matchHeading`, currently this pull request uses `match`.

### CODE REVIEW

**Comment by wooorm** [comment]
similar feedback as https://github.com/remarkjs/remark-contributors/pull/17#issuecomment-782701549!

**Comment by nnmrts** [comment]
Renamed it to heading (forgot to mention that in the commit message), added string support and removed the regex wrapping to make it case-insensitive, because `mdast-util-heading-range` already does that.

File: index.js [hunk 1]
```diff
@@ -23,10 +23,12 @@ var https = 'https://'
 function license(options) {
   var settings = options || {}
   var finals = settings.ignoreFinalDefinitions
+  var test = settings.heading || licenseHeadingRegexp
+
   var headingOptions = {
     ignoreFinalDefinitions:
       finals === undefined || finals === null ? true : finals,
-    test: licenseHeadingRegexp
+    test
```
Comment (wooorm) [inline index.js]:
```suggestion
    test: test
```

Just to be careful and not break anyone for now

**Review by wooorm — COMMENTED** [review]
Sweet, thanks. One tiny nit!

File: index.js (hunk 1 above)
Comment (nnmrts) [reply index.js]: Alright, committed it!

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
[Released](https://github.com/remarkjs/remark-license/releases/tag/5.2.0)!

**Pull request merged**